### ISSUE TITLE
Lower performance A-Frame 1.0.x vs 0.9.2

### ISSUE THREAD

**Issue by smeybi** [aframevr/aframe#4439]
Hi,
I found out the performance is noticeably lower (specially with the Quest) since A-Frame Version 1.0.x vs. 0.9.2

I also tested with the Vive and Firefox, the same, lower performance. Unfortunately I have not found the reason.

Here some tests:

https://glitch.com/~aframe-performancetest-v092

https://glitch.com/~aframe-performancetest-v104

**Comment by rigel-prime** [comment]
Yes, I'm seeing this as well in my projects (tested on Quest w/ Oculus Browser)

**Comment by dmarcos** [comment]
There are no major API changes from 0.9.2 to 1.0.4. It needs profiling. I haven't noticed any perf regression on our deployed experiences.

**Comment by IvoJager** [comment]
This is a duplicate of #4347. 
It is because of changes to the standard material shaders in THREE.js.
We're sticking with a 0.9.2 and 1.0 hybrid as we have found 1.0+  to be too slow to be usable for our purposes.

**Comment by dmarcos** [comment]
@IvoJager Thanks! If it's the same issue, switching to `flat` shader should work around the problem, right?

**Comment by IvoJager** [comment]
@dmarcos Yep - if it's the same issue it should! :)

**Comment by arpu** [comment]
make sure alpha is set true in the renderer, there was some problems with set it false on oculus browser

**Comment by arpu** [comment]
https://github.com/mrdoob/three.js/issues/18338

**Comment by dmarcos** [comment]
Anything actionable here on A-Frame side?

**Comment by IvoJager** [comment]
In our case, we've been able to isolate the performance drop (across other headsets as well - e.g. Vive Focus) to the changes in the standard materials in three.js. I believe it is the A-Frame team's intention #4347 to implement three.js' other, simpler materials that have a lower, fixed computation costs for the 1.1 milestone. This should avoid coding against a moving target (e.g. upstream three.js changes to the standard material) in the future, while allowing us to be more clever with resource allocation at little expense to visual fidelity.

**Comment by arpu** [comment]
the oculusGo alpha problem is fixed in upstream OculusBrowser

@IvoJager  can you share the standard material change in threejs?

**Comment by IvoJager** [comment]
@arpu  The biggest change is in the integrateSpecularBRDF shader function. I believe it has to do with the way roughness is rendered. I am not an expert in this subject matter though.
